examples/balanced_lineupsNBA.py

```diff
@@ -0,0 +1,118 @@
+"""
+Generate as many lineups as you want with
+max exposures of your selected exposure to any one player,
+with a specified minimum average of your choice.
+The output here is a CSV file that can be used
+to upload lineups at https://www.draftkings.com/lineup/upload
+
+Pre-reqs:
+1) Salary file exists at passed location
+2) Projection file exists at passed location
+3) PIDS file downloaded from aforementioned location
+
+Example usage:
+```python
+from examples import balanced_lineupsNBA
+balanced_lineupsNBA.run()
+```
+"""
+
+import random
+import numpy
+from collections import Counter
+from optimize import run as optimizer_run
+from argparse import Namespace
+
+from csv_upload.nba_upload import (
+    update_upload_csv,
+    create_upload_file,
+    map_pids,
+)
+
+DEFAULT_ARGS = dict(
+    dtype='wr',
+    duo='n',
+    s='n',
+    w=5,
+    season=2017,
+    historical='n',
+    i=1,
+    league='NBA',
+    limit='n',
+    no_double_te='y',
+    lp=0,
+    mp=500,
+    ms=100000,
+    sp=1000,
+    banned=[],
+    po=0,
+    pids='data/pid-fileNBA.csv',
+    salary_file='data/current-salaries.csv',
+    projection_file='data/current-projections.csv',
+    home=None,
+    v_avg=100,
+    source='nba_rotogrinders',
+    flex_position=None,
+    locked=None,
+    teams=None,
+    po_location=None,
+    min_avg=None,
+)
+
+
+def is_duplicate(r, rp):
```

Comment:
We should probably abstract some of the examples stuff like the exposure printing and `is_duplicate`. This is only if you feel like it though, not critical review comment 👍